addons/utm/models/utm_medium.py

```diff
@@ -49,14 +51,15 @@ def _unlink_except_utm_medium_record(self):
                 ))
 
     def _fetch_or_create_utm_medium(self, name, module='utm'):
+        name_normalized = re.sub(r"[\s|.]", "_", name.lower())
```

Comment:
Should come with unit tests :) to put into test_utm.py probably, after test_find_or_create_record_case (it seems _fetch_or_create_utm_medium is currently not tested so good opportunity to do it)

Reply:
Idea for that kind of test: create one existing data for the test, check the fetch or create with name with some variations, check we always fetch the right one - then check for a medium to create, same with some variations.